Runtime/Interaction/SnapZone.cs

```diff
@@ -317,16 +340,31 @@ public override void ProcessInteractor(XRInteractionUpdateOrder.UpdatePhase upda
             {
                 base.ProcessInteractor(updatePhase);
             }
-
+            
             if (socketActive)
```

Comment:
invert if and return early.

Reply:
Can you split this method in two, one for handling highlights and other for handling snapping?